packages/querygrid/src/components/user/UserDetailHeader.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+import { Map } from 'immutable'
+import { User } from "@glass/base";
+import { getUserLastLogin, getUserPermissionsDisplay } from "./actions";
+import { PageDetailHeader } from "../forms/PageDetailHeader";
+
+interface HeaderProps {
+    title: string
+    user: User
+    userProperties: Map<string, any>
+    dateFormat: string
+    renderButtons?: () => any
+}
+
+export class UserDetailHeader extends React.Component<HeaderProps> {
+
+    render() {
+        const { user, userProperties, title, dateFormat, renderButtons } = this.props;
+        const lastLogin = getUserLastLogin(userProperties, dateFormat);
+
+        return (
+            <PageDetailHeader
+                user={user}
+                iconUrl={user.avatar}
+                title={title}
+                description={getUserPermissionsDisplay(user).join(', ')}
+                leftColumns={10}
+            >
+                {lastLogin && <div className={'detail__header--desc'}>Last Login: {lastLogin}</div>}
+                {renderButtons && <div className={lastLogin ? 'detail__header--buttons' : ''}>{renderButtons()}</div>}
```

Comment:
Why does the style here depend on whether the user has logged in previously?

Reply:
For button placement. If there is no last login text, then the buttons go at the top right. If there is text, then give some space below the text before rendering the buttons.